api.py

```
@@ -23,33 +23,47 @@
-    url=url, username=username, password=password, refresh_schema=False
-)
-create_vector_index(neo4j_graph)
+# Initialize Neo4j graph
+try:
+    neo4j_graph = Neo4jGraph(
+        url=url, username=username, password=password, refresh_schema=False
+    )
+    create_vector_index(neo4j_graph)
+except Exception as e:
+    logger.error(f"Error initializing Neo4j graph: {e}")
+    raise HTTPException(status_code=500, detail="Error initializing Neo4j graph")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)